comp/metadata/hostgpu/impl/hostgpu.go

```diff
@@ -152,6 +152,10 @@ func (gh *gpuHost) fillData() {
 func (gh *gpuHost) getPayload() marshaler.JSONMarshaler {
 	gh.fillData()
 
+	if gh.data.Devices == nil || len(gh.data.Devices) == 0 {
```

Comment:
`len` of a nil slice will return 0:
```suggestion
	if len(gh.data.Devices) == 0 {
```

Reply:
i intented to check the `gh.data == nil`, fixed

Reply:
the struct is private and the data field is always initialized in the ctor, removing that check